tests/utils/test_environment.py

```diff
@@ -50,3 +54,41 @@ def test_mlflow_conda_env_includes_pip_dependencies_and_pip_is_specified(pip_spe
         assert conda_dep in env["dependencies"]
     assert pip_specification in env["dependencies"]
     assert env["dependencies"].count("pip") == (2 if pip_specification == "pip" else 1)
+
+
+def test_resolve_pip_requirements(tmpdir):
+    assert _resolve_pip_requirements(["a", "b"]) == ["a", "b"]
+    assert _resolve_pip_requirements(["a==0.0", "b>1.1"]) == ["a==0.0", "b>1.1"]
```

Comment:
The current implementation doesn't detect conflicting requirements:

```python
log_model(pip_requirements=["pandas==1.2.3", "pandas==4.5.6"])
```

Not sure if this is really necessary, we could detect them and raise an error / warning.

Reply:
Tested how pip works for conflicting requirements:

```
% cat a.txt
scikit-learn==0.24.1
scikit-learn==0.24.2

% pip install -r a.txt
Requirement already satisfied: scikit-learn==0.24.1 in /Users/harutakakawamura/.pyenv/versions/miniconda3-4.7.12/envs/mlflow-dev-env/lib/python3.7/site-packages (from -r a.txt (line 1)) (0.24.1)
ERROR: Cannot install scikit-learn==0.24.1 and scikit-learn==0.24.2 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested scikit-learn==0.24.1
    The user requested scikit-learn==0.24.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
```

Looks like it throws